functions/Set-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,332 @@
+
+.EXAMPLE   
+Set-DbaAgentJobStep -SqlServer 'sql' -JobName 'Job1' -StepName 'Step1' -Database 'msdb'
+Changes the database of the step in "Job1" with the name "Step1" to 'msdb'
+
+.EXAMPLE   
+"sql1", "sql2", "sql3" | Set-DbaAgentJobStep -JobName 'Job1' -StepName 'Step1' -Database 'msdb'
+Changes the database of the step in "Job1" with the name "Step1" to 'msdb' for multiple servers
+
+#>   
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+
+    param(
+        [parameter(Mandatory = $true, ValueFromPipeline=$true)]
+		[object[]]$SqlServer,
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$JobName,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$StepName,
+        [Parameter(Mandatory = $false)]
+        [string]$NewName,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('ActiveScripting','AnalysisCommand','AnalysisQuery','CmdExec','Distribution','LogReader','Merge','PowerShell','QueueReader','Snapshot','Ssis','TransactSql')]
+        [string]$Subsystem,
+        [Parameter(Mandatory = $false)]
+        [string]$Command,
+        [Parameter(Mandatory = $false)]
+        [int]$CmdExecSuccessCode,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('QuitWithSuccess', 'QuitWithFailure', 'GoToNextStep', 'GoToStep')]
+        [string]$OnSuccessAction,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($OnSuccessAction -ne 'GoToStep') -and ($_ -ge 1))
+            {
+                Throw "Parameter OnSuccessStepId can only be used with OnSuccessAction 'GoToStep'."
+            }
+            else 
+            {
+                $true    
+            }
+        })]
+        [int]$OnSuccessStepId,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('QuitWithSuccess', 'QuitWithFailure', 'GoToNextStep', 'GoToStep')]
+        [string]$OnFailAction,
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({
+            if(($OnFailAction -ne 'GoToStep') -and ($_ -ge 1))
+            {
+                Throw "Parameter OnFailStepId can only be used with OnFailAction 'GoToStep'."
+            }
+            else 
+            {
+                $true    
+            }
+        })]
+        [int]$OnFailStepId,
+        [Parameter(Mandatory = $false)]
+        [string]$DatabaseName,
+        [Parameter(Mandatory = $false)]
+        [string]$DatabaseUserName,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryAttempts,
+        [Parameter(Mandatory = $false)]
+        [int]$RetryInterval,
+        [Parameter(Mandatory = $false)]
+        [string]$OutputFileName,
+        [Parameter(Mandatory = $false)]
+        [string]$ProxyName,
+        [switch]$Silent
+    )
+
+    PROCESS
+    {
+        # Try connecting to the instance
```

Comment:
Since you accept multiple instances as parameter, be sure to allow for non-pipeline input by adding a foreach loop:
```
foreach ($instance in $SqlInstance)
{
    # ...
}
```